backend/entities/auth/base/service.go

```diff
@@ -120,6 +126,24 @@ func (a *AuthService) Register(userBody usersEntities.CreateUserRequestBody) (*m
 }
 
 func (a *AuthService) Refresh(refreshToken string) (*auth.Token, error) {
+	isActive, err := a.Stores.ActiveToken.IsActive(refreshToken)
+	if err != nil {
+		return nil, err
+	}
+
+	if !isActive {
+		return nil, utilities.Unauthorized()
+	}
+
+	isblacklisted, err := a.Stores.Blacklist.IsTokenBlacklisted(refreshToken)
+	if err != nil {
+		return nil, err
+	}
+
+	if isblacklisted {
+		return nil, fmt.Errorf("refresh token is blacklisted")
```

Comment:
what make sense as an error code here? if it isn't 500, what would it be? maybe create a helper in utilities/api_error similar to utilities.Unauthorized()